File: macos-widget/PlaneTrackerWidget/Helpers/WMOCodes.swift
import Foundation

enum WMOCodes {
    /// WMO weather code → SF Symbol name
    static func sfSymbol(for code: Int) -> String {
        switch code {
        case 0:           return "sun.max.fill"
        case 1:           return "sun.min.fill"
        case 2:           return "cloud.sun.fill"
        case 3:           return "cloud.fill"
        case 45, 48:      return "cloud.fog.fill"
        case 51, 53:      return "cloud.drizzle.fill"
        case 55, 61, 63:  return "cloud.rain.fill"
        case 65:          return "cloud.heavyrain.fill"
        case 71, 73, 77:  return "cloud.snow.fill"
        case 75:          return "snowflake"
        case 80, 81:      return "cloud.rain.fill"
        case 82:          return "cloud.bolt.rain.fill"
        case 85, 86:      return "cloud.snow.fill"
        case 95, 96, 99:  return "cloud.bolt.fill"
        default:          return "cloud.fill"
        }
    }

    /// WMO weather code → short description
    static func description(for code: Int) -> String {
        switch code {
        case 0:           return "Clear"
        case 1:           return "Mostly Clear"
        case 2:           return "Partly Cloudy"
        case 3:           return "Overcast"
        case 45, 48:      return "Foggy"
        case 51, 53:      return "Drizzle"
        case 55:          return "Heavy Drizzle"
        case 61:          return "Light Rain"
        case 63:          return "Rain"
        case 65:          return "Heavy Rain"
        case 71:          return "Light Snow"
        case 73:          return "Snow"
        case 75:          return "Heavy Snow"
        case 77:          return "Snow Grains"
        case 80, 81:      return "Showers"
        case 82:          return "Heavy Showers"
        case 85, 86:      return "Snow Showers"
        case 95:          return "Thunderstorm"
        case 96, 99:      return "Hailstorm"
        default:          return "Unknown"
        }
    }
}

File: android-widget/app/src/main/java/com/planetracker/util/WMOCodes.kt
package com.planetracker.util

object WMOCodes {
    /** WMO weather code → emoji */
    fun emoji(code: Int): String = when (code) {
        0 -> "\u2600\uFE0F"          // ☀️
        1 -> "\uD83C\uDF24\uFE0F"    // 🌤️
        2 -> "\u26C5"                 // ⛅
        3 -> "\u2601\uFE0F"          // ☁️
        45, 48 -> "\uD83C\uDF2B\uFE0F" // 🌫️
        51, 53 -> "\uD83C\uDF26\uFE0F" // 🌦️
        55, 61, 63 -> "\uD83C\uDF27\uFE0F" // 🌧️
        65 -> "\uD83C\uDF27\uFE0F"   // 🌧️
        71, 73, 77 -> "\uD83C\uDF28\uFE0F" // 🌨️
        75 -> "\u2744\uFE0F"          // ❄️
        80, 81 -> "\uD83C\uDF27\uFE0F" // 🌧️
        82 -> "\u26C8\uFE0F"          // ⛈️
        85, 86 -> "\uD83C\uDF28\uFE0F" // 🌨️
        95, 96, 99 -> "\u26C8\uFE0F"  // ⛈️
        else -> "\uD83C\uDF21\uFE0F"  // 🌡️
    }

    /** WMO weather code → short description */
    fun description(code: Int): String = when (code) {
        0 -> "Clear"
        1 -> "Mostly Clear"
        2 -> "Partly Cloudy"
        3 -> "Overcast"
        45, 48 -> "Foggy"
        51, 53 -> "Drizzle"
        55 -> "Heavy Drizzle"
        61 -> "Light Rain"
        63 -> "Rain"
        65 -> "Heavy Rain"
        71 -> "Light Snow"
        73 -> "Snow"
        75 -> "Heavy Snow"
        77 -> "Snow Grains"
        80, 81 -> "Showers"
        82 -> "Heavy Showers"
        85, 86 -> "Snow Showers"
        95 -> "Thunderstorm"
        96, 99 -> "Hailstorm"
        else -> "Unknown"
    }
}
